feat: add ci-cd example for upload-all-bundles

danlordan opened the pull request:
Add ci example for managing flow resources under version control

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+          repository: ucroo/ihub-developer-scripts
+          path: ihub-developer-scripts
+
+      - name: Install jq
```
I think jq comes installed on gh workers by default

danlordan replied:
leaving as is in case anyone running locally

Commit:
feat: add CI/CD tutorial for Flow bundle deployment

Added comprehensive tutorial demonstrating best practices for automating Flow bundle deployment using GitHub Actions with automatic multi-bundle discovery and upload.

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+          sudo apt-get update && sudo apt-get install -y jq
+          which jq || echo "jq not found in PATH"
+
+      - name: Set up credentials directory
```
If we don't have these values, I'd assume we would want the job to fail here

danlordan replied:
Good call, implementing

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+          if [ -n "${{ secrets.API_HOST }}" ]; then
+            echo "${{ secrets.API_HOST }}" > /home/runner/creds/${SCHOOL_ALIAS}.api
+          fi
+          if [ -n "${{ secrets.CURL_ARGS }}" ]; then
```
What do we expect curl args to be?

danlordan replied:
This is used in `uploadBundle.sh` in ihub-developer-scripts, leaving as is so documentation can stay there

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+        run: |
+          set -e  # Exit on any error
+
+          if [ ! -d "${BUNDLES_DIR}" ]; then
```
I believe this will always fail? Pretty sure each step runs in an isolated process so you'll need to set this value out the gate

schroeder-g replied:
Isn't bundlesdir gonna be constant, anyhow?

danlordan replied:
I actually like hardcoding it for the example, instead of having it configurable like it is now, updating that.

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+            fi
+          done
+        env:
+          PATH: ${{ github.workspace }}/ihub-developer-scripts:/usr/bin:$PATH
```
I'm pretty sure that with this you can reference the uploadBundle.sh script without a relative path

danlordan replied:
Excellent callout, updated

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+                if [ -z "$bundle_name" ] || [ "$bundle_name" = "null" ]; then
+                  echo "Could not find bundle ID in JSON file $json_file, falling back to filename"
+                  # Fallback: use directory name or basic filename cleaning
+                  bundle_name=$(echo "$raw_bundle_name" | sed 's/^v1_US_//' | sed 's/_US_/_/g')
```
Idk what this means. Are these the only prefixes we need to trim?

danlordan replied:
Changed the handling here

schroeder-g commented on flowBundlesCICD/README.md:
```diff
@@ -0,0 +1,159 @@
+
+## Disclaimer
+
+These best practices are provided as guidance and examples. While we aim to provide secure and reliable patterns, the ultimate responsibility for your deployment environment, security, and operational practices is your own. Please review, test, and adapt these workflows according to your organization's specific requirements and security policies.
```
IMO—makes sense to include an action like this by default when we set up the ucroo community

danlordan replied:
Yeah we should add this to the top level of tutorials, will make a note

schroeder-g commented on flowBundlesCICD/README.md:
```diff
@@ -0,0 +1,159 @@
+| Secret       | Description               | Required    |
+| ------------ | ------------------------- | ----------- |
+| `FLOW_TOKEN` | Flow access token         | ✅ Required |
+| `FLOW_HOST`  | Flow host URL             | Optional    |
```
Maybe I missed it but I would expect this to be required?

danlordan replied:
No, this is optional only if your flow doesn't follow standard naming conventions, will add a note on default

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+          cd "${BUNDLES_DIR}"
+
+          # Find all bundle directories
+          for bundle_dir in */; do
```
Just from a code hygiene perspective it could be nice to move these scripts into .github/scripts and call them here

danlordan replied:
Good call, leaving for future

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
```
How does this work?

danlordan replied:
On pushes to main, we run the dispatch

schroeder-g commented on flowBundlesCICD/.github/workflows/upload-all-bundles.yml:
```diff
@@ -0,0 +1,113 @@
+# - Uploads each bundle using uploadBundle.sh
+#
+# Configuration:
+# - Set SCHOOL_ALIAS to your school's identifier (e.g., myschool, staging)
```
How do schools know their school identifier?

danlordan replied:
This is actually unneeded since we're doing single env management, removing completely for ease of use

schroeder-g reviewed (commented):
This looks rad!! I have some clarifying qs

Commit:
feat: add ci-cd example for upload-all-bundles
